source/Applications/Advanced/HandEyeCalibration/PoseConversions/PoseConversions.cpp

```diff
@@ -11,6 +11,7 @@ This example shows how to use Eigen to convert to and from:
 #include <Eigen/Core>
 #include <Eigen/Dense>
 #include <Eigen/Geometry>
+#include <Zivid/Zivid.h>
```

Comment:
Is the order of includes correct? Did you run clang-tidy?

Reply:
clang-tidy doesn't understand which headers are internal or not, it just re-orders each group of headers alphabetically. So if you do this, they will be re-ordered with internal headers at the bottom (wrongly):

```c++
#include <Zivid/Zivid.h>
#include <Eigen/Core>
#include <Eigen/Dense>
#include <Eigen/Geometry>
```

But if you do it like this, it will only re-order each group of headers internally, and leave the internal header at the top (correctly):

```c++
#include <Zivid/Zivid.h>

#include <Eigen/Core>
#include <Eigen/Dense>
#include <Eigen/Geometry>
```

Reply:
Done. For this and other files.

Reply:
> clang-tidy doesn't understand which headers are internal or not

`clang-format` can do this pretty well if you use `IncludeBlocks::IBS_Regroup` together with `IncludeCategories`, we just have not done that yet in SDK. Iirc it also understand what is system headers and what is yours, if you use it together with a compilation database, but that does not work if you run it from inside Visual Studio.